packages/terminal/src/browser/base/terminal-service.ts

```diff
@@ -46,6 +46,12 @@ export interface TerminalService {
      */
     getById(id: string): TerminalWidget | undefined;
 
+    /**
+     * @param id - the terminal id (NOT the terminal widget id!)
+     * @return the widget
+     */
+    getByTerminalId(terminalId: number): TerminalWidget | undefined;
```

Comment:
I think adding a new method to interface should be mentioned as `breaking changes` in `changelog`

Reply:
we can, but I don't think anyone implements it. This interface should be removed and clients should use `TerminalFrontendContribution` directly.

It is unnecessary boilerplate and we don't encourage it: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#classes-over-interfaces

Also we don't require a breaking change notice only if there is a sufficient adoption for experimental API:  https://github.com/eclipse-theia/theia/blob/master/doc/api-management.md#experimental I believe there is only one implementation `TerminalFrontendContribution`.

Reply:
@elaihau Could you file an issue to remove `TheiaWidget` and `TheiaService` interfaces and rename `TerminalWidgetImpl` to `TerminalWidget`. We could do it via deprecation for clients accessing it, i.e. existing interfaces can extend `TerminalWidget` and `TerminalFrontendContribution` and tell to use them instead.

Reply:
@akosyakov 
> we can, but I don't think anyone implements it.

I don't know if anyone implements it...

> It is unnecessary boilerplate and we don't encourage it

I totally agree with you, but at the moment there is an interface and, as I mentioned above, I don't know if anyone implements it. I pointed out that, but at your discretion...

> Also we don't require a breaking change notice only if there is a sufficient adoption for experimental API

thanks for the explanation!

Reply:
> @elaihau Could you file an issue to remove `TheiaWidget` and `TheiaService` interfaces 

@akosyakov 
Just want to confirm with you: you meant `TerminalWidget` and `TerminalService`, did you?

Reply:
yes, sorry 🤦‍♂